builder/pipeline_test.cc

```diff
@@ -1,15 +1,71 @@
 #include <gtest/gtest.h>
 
 #include <cassert>
+#include <fstream>
 
 #include "builder/pipeline.h"
+#include "builder/replica_pipeline.h"
 #include "runtime/expr.h"
 #include "runtime/pipeline.h"
 #include "runtime/thread_pool.h"
 #include "runtime/visualize.h"
+#include "tools/cpp/runfiles/runfiles.h"
 
 namespace slinky {
 
+namespace internal {
+extern void check_replica_pipeline(const std::string& name, const std::string& replica_text,
+    span<const raw_buffer*> inputs, span<const raw_buffer*> outputs, const std::string& test_src);
+}  // namespace internal
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+std::string read_entire_file(const std::string& pathname) {
+  std::ifstream f(pathname, std::ios::in | std::ios::binary);
+  std::string result;
+
+  f.seekg(0, std::ifstream::end);
+  size_t size = f.tellg();
+  result.resize(size);
+  f.seekg(0, std::ifstream::beg);
+  f.read(result.data(), result.size());
+  if (!f.good()) {
+    std::cerr << "Unable to read file: " << pathname;
+    std::abort();
+  }
+  f.close();
+  return result;
+}
+
+class PipelineTest : public testing::Test {
+protected:
+  void SetUp() override {
```

Comment:
It looks like there could just be a free function `read_replica_pipeline_test_cc()` (no need to cache it IMO) and then we don't need test fixtures.

Reply:
Re-reading the same text file multiple times seems gratuitously wasteful, and test fixtures seem very benign. I'll rewrite it to add back the overhead if you prefer, but I don't agree with the decision.

Reply:
This would be a really easy thing to wrap with a simple static initializer:

```
std::string get_replica_golden() {
  static std::string golden = [](){ ... }();
  return golden;
}
```
We talked about this offline, but just in case anyone else is following: I'd rather avoid test fixtures because I feel like I have to read them very carefully to make sure they aren't sharing state across tests, which is something to be avoided IMO.